int8     detected
float32  x1
float32  y1
float32  z1
float32  x2
float32  y2
float32  z2